src/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import type { HTMLProps, ReactNode } from "react";
+
+import Field from "../Field";
+
+import type { PropsWithSpread } from "types";
+
+export type Props = PropsWithSpread<
+  {
+    /**
+     * The label name for the switch
+     */
+    label: ReactNode;
+    /**
+     * Whether the switch is disabled or not
+     */
+    disabled?: boolean;
+  },
+  HTMLProps<HTMLInputElement>
```

Comment:
Looks like you allow passing additional props to the component (input element props), but you don't really read them or pass them to the input in the component?